model/task/task.go

```diff
@@ -2558,35 +2577,67 @@ func ArchiveMany(tasks []Task) error {
 	defer session.EndSession(ctx)
 
 	txFunc := func(sessCtx mongo.SessionContext) (interface{}, error) {
-		oldTaskColl := evergreen.GetEnvironment().DB().Collection(OldCollection)
-		_, err = oldTaskColl.InsertMany(ctx, archived)
-		if err != nil {
-			return nil, err
+		var err error
+		if len(toArchive) > 0 {
+			oldTaskColl := evergreen.GetEnvironment().DB().Collection(OldCollection)
+			_, err = oldTaskColl.InsertMany(sessCtx, toArchive)
+			if err != nil {
+				return nil, errors.Wrap(err, "archiving tasks")
+			}
 		}
-
-		taskColl := evergreen.GetEnvironment().DB().Collection(Collection)
-		_, err = taskColl.UpdateMany(ctx, bson.M{
-			IdKey: bson.M{
-				"$in": taskIds,
-			},
-		},
-			bson.M{
-				"$unset": bson.M{
-					AbortedKey:   "",
-					AbortInfoKey: "",
-				},
-				"$inc": bson.M{
-					ExecutionKey: 1,
+		if len(tasksIds) > 0 {
+			// TODO (EVG-17322): Remove RunCommand call and replace with aggregate + merge
+			result := evergreen.GetEnvironment().DB().RunCommand(sessCtx,
+				bson.D{
+					bson.E{Key: "update", Value: Collection},
+					bson.E{Key: "updates", Value: bson.A{bson.M{
+						"q": bson.D{{Key: "_id", Value: bson.D{{Key: "$in", Value: tasksIds}}}},
```

Comment:
would it be possible to build the aggregation like this and/or have more comments on it? 
https://github.com/evergreen-ci/evergreen/blob/1b36641d314cbf165cb1e36bb917a1a8d627520b/model/task_history.go#L559

Reply:
Yes, I've added some comments and I think it helps explain the three steps in the pipeline very well!